profiles/lib/handlers.lua

```diff
@@ -399,15 +399,22 @@ function Handlers.handle_oneway(way,result,data,profile)
   end
 end
 
+function Handlers.handle_weights(way,result,data,profile)
+  if properties.weight_name == 'distance' then
+    -- set weight rates to 1 for the distance weight, edge weights are distance / rate
+    result.forward_rate = 1
```

Comment:
agree, i added checks and also set `result.weight` explicitly to 0, that means "use rate values to compute edge weights"